### PR TITLE
HDFS: support macro in filename

### PR DESCRIPTION
**Pull request by pzoleex**
Don't merge it, needs more test

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

File: modules/java-modules/hdfs/src/main/java/org/syslog_ng/hdfs/HdfsDestination.java [hunk 1]
```diff
@@ -141,53 +147,57 @@ public boolean send(LogMessage logMessage) {
...
+    
+    private FSDataOutputStream getFSDataOutputStream(String resolvedFileName) {
+        HdfsFile hdfsFile = hmapOpenedFiles.get(resolvedFileName);
 
-    private String truncateFileName(String fileName, int maxLength) {
-        if (fileName.length() > maxLength) {
-            return fileName.substring(0, maxLength);
+        if (hdfsFile == null) {
+            hdfsFile = createHdfsFile(resolvedFileName);
+            hmapOpenedFiles.put(resolvedFileName, hdfsFile);
```
Comment (lbudai) [inline modules/java-modules/hdfs/src/main/java/org/syslog_ng/hdfs/HdfsDestination.java]: just call it `openedFiles`

File: modules/java-modules/hdfs/src/main/java/org/syslog_ng/hdfs/HdfsDestination.java [hunk 2]
```diff
@@ -196,16 +206,23 @@ private boolean isDfsSupportAppendEnabled(FileSystem hdfs) {
...
-            } catch (IOException e) {
-                logger.debug(String.format("Flush failed, reason: %s", e.getMessage()));
+        /*
+         * Trying to flush however the effect is questionable... (It is not guaranteed
+         * that data has been flushed to persistent store on the datanode)
+         */
+
+        Iterator<Entry<String, HdfsFile>> it = hmapOpenedFiles.entrySet().iterator();
+        logger.debug("Flushing hdfs");
+        while (it.hasNext()) {
```
Comment (lbudai) [inline modules/java-modules/hdfs/src/main/java/org/syslog_ng/hdfs/HdfsDestination.java]: use the for each loop

**Comment by lbudai** [comment]
and could you re-format the code in a separate patch ? 
I think it is time to fix this tab/space issue :)

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Comment by kira-syslogng** [comment]
Build SUCCESS, the tests were executed on test branch: master and test suite: functions

**Pull request merged**